cadc-storage-permissions/src/main/java/org/opencadc/inventory/permissions/ReadWriteGrant.java

```diff
@@ -0,0 +1,109 @@
+ *  written permission.                  préalable et particulière
+ *                                       par écrit.
+ *
+ *  This file is part of the             Ce fichier fait partie du projet
+ *  OpenCADC project.                    OpenCADC.
+ *
+ *  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+ *  you can redistribute it and/or       vous pouvez le redistribuer ou le
+ *  modify it under the terms of         modifier suivant les termes de
+ *  the GNU Affero General Public        la “GNU Affero General Public
+ *  License as published by the          License” telle que publiée
+ *  Free Software Foundation,            par la Free Software Foundation
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  $Revision: 4 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.inventory.permissions;
+
+import java.net.URI;
+import java.util.Date;
+import java.util.List;
+import org.opencadc.gms.GroupURI;
+
+/**
+ * Holds grant information about an artifact.
+ * 
+ * @author majorb
+ *
+ */
+public class ReadWriteGrant extends Grant {
+
+    /**
+     * Construct a grant for the given artifactURI.
+     * @param artifactURI The applicable targetURI.
+     */
+    public ReadWriteGrant(URI artifactURI) {
+        super(artifactURI);
+    }
+
+    /**
+     * Get the read-write group list.
+     * @return The read-write groups.
+     */
+    public List<GroupURI> getGroups() {
+        return groups;
+    }
+
+    /**
+     * Get date after which the Grant is considered expired and should be renewed.
+     * @return The grant expiry date.
+     */
+    public Date getExpiryDate() {
+        return expiryDate;
+    }
```

Comment:
Why is getExpiryDate() not just implemented in Grant instead of being abstract?

Reply:
I wasn't sure how it is determined? If it's a set period of time then it could be in Grant. Or is there another way to calculate and set it based on the archive when getting the permissions? Is it the same for read and write?

Reply:
I don't know what concept it is trying to portray.  I originally had an 'evaluation date' field where my intent was that clients could see this date to determine if the information was recent enough to be considered correct.

Reply:
Same idea but a more explict name to indicate the service doesn't consider the grant reliable past that time and it should be renewed. Saying the grant has now `expired` is a strong hint to renew it.